launch/launch/main.py

```diff
@@ -21,28 +21,45 @@
 from launch.loader import load_launch_file
 
 
-def file_exists(filename):
-    if not os.path.exists(filename) or not os.path.isfile(filename):
-        raise argparse.ArgumentError("'%s' does not exist" % filename)
-    return filename
-
-
 def main(argv=sys.argv[1:]):
     parser = argparse.ArgumentParser(
         description='Launch the processes specified in a launch file.')
     parser.add_argument(
         'launch_file',
-        type=file_exists,
+        type=str,
         nargs='+',
         help='The launch file.')
+    parser.add_argument(
+        'arg',
+        type=str,
+        nargs='*',
+        help='An argument to the launch file (e.g., arg_name:=value).')
     args = parser.parse_args(argv)
 
-    arguments = {}
+    # Get the list of launch files passed on the command line
+    # NOTE: since both the launch files and launch arguments are consecutive
+    # lists of arguments, argparse will put all arguments into the launch_file
+    # list, and the arg list will be empty. This is because the '+' list is
+    # greedy and consumes the remaining arguments. Therefore the arguments
+    # need to be validated afterward
+    launch_files = []
+    for arg in args.launch_file:
+        # Store consecutive existing files and stop once a
+        # non-existing file is found
+        if os.path.isfile(arg):
+            launch_files.append(arg)
+        else:
+            break
+
+    # Ensure that at least one existing launch file was given
+    if len(launch_files) == 0:
+        parser.error('you must pass at least one valid launch file')
+        sys.exit(2)
```

Comment:
This line is never executed. [parser.error](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error) already does exactly the same.